### DISCUSSION TITLE
FR: move dropdown arrow next to "set state"

### DISCUSSION THREAD

**Discussion by Mariusthvdb** [home-assistant/frontend#10915] [Dashboard]
<img width="2128" alt="Schermafbeelding 2021-12-14 om 14 09 21" src="https://user-images.githubusercontent.com/33354141/146004494-bd275144-7feb-403f-a852-f27f9121b6b3.png">

speaks for itself I guess. As a matter of fact, I hadn't even noticed it, and it was only because of reading the community post by Ildar I went over the right side of the window to discover it ;-)

Please consider moving it?

secondly, please auto-dropdown, if an entity_id is clicked in the entity column. Currently, clicking an entity does nothing, which really feels odd.

so this is not related to any theme, a screenshot for core default theme:
<img width="2124" alt="Schermafbeelding 2021-12-14 om 14 21 54" src="https://user-images.githubusercontent.com/33354141/146012754-409c28a3-2dc0-42c6-a2af-16659453bf07.png">

as @Petro31 suggests: https://community.home-assistant.io/t/2021-12-new-configuration-menu-the-button-entity-and-gorgeous-area-cards/365680/384

thanks

**Comment by spacegaier** [comment]
Counter proposal: Making the collapsible section smaller and not use the same width as the table below, feels weird to me. Instead I would outline the section as we do e.g. with the attribute section in the more-info. I will create a PR for this proposal (https://github.com/home-assistant/frontend/pull/10917).

**Reply by matthiasdebaat** [reply]
Yeah, we should make it smaller

**Reply by spacegaier** [reply]
Which would be the opposite of my proposal 😆 .

My proposal is having it outlined, like this:

![image](https://user-images.githubusercontent.com/114137/146028848-d4596da6-edbc-4436-a32f-3e3dad3f6b04.png)

Feels visually cleaner to me.

**Reply by matthiasdebaat** [reply]
Like that even better!

**Comment by Mariusthvdb** [comment]
and what do you think about the auto fold-down? in that case we would not even need the Set state box in the first place maybe?

Clicking an entity would simply reveal the Set state box in open state

btw, Ive never felt it to be related to the width of the full table below. Ive always thought it to be a separate section, so, adjusting the width for the Set state header and down arrow to the fist column of the entity would not feel strange at all.

as a matter of fact, everything is delimited to that width currently, except for the down arrow icon...
<img width="711" alt="Schermafbeelding 2021-12-14 om 16 53 39" src="https://user-images.githubusercontent.com/33354141/146032772-bc68d320-29d7-4c2f-bc7b-e237a8cc859e.png">

populated, everything still nicely spread over the first column width:

<img width="757" alt="Schermafbeelding 2021-12-14 om 16 56 44" src="https://user-images.githubusercontent.com/33354141/146033445-9f3cdf58-5df4-48f9-a01f-7be276ace661.png">

while at the same time:
<img width="2213" alt="Schermafbeelding 2021-12-14 om 16 57 16" src="https://user-images.githubusercontent.com/33354141/146033563-1015dfee-049a-4af0-860e-2143541be582.png">

Honestly think having it outlined would only make it stand out even more, accenting the oddity, instead of solving it. 

Maybe renaming Set state to More-info, or details would help?. In which dropped-down view you can set state at the bottom

**Reply by spacegaier** [reply]
In my setup the state section is always bigger than the first table column, actually the first two.

![image](https://user-images.githubusercontent.com/114137/146036502-59394b13-980f-48f1-9324-403506af0a69.png)

I like the auto-expand idea. Added it to my PR.

**Reply by Mariusthvdb** [reply]
thanks, appreciated!

guess the screen real estate makes that all happen, used to my wide screen the attributes column is ridiculously empty too ;-) While on the iPhone, no attributes are available at all, not even optional.

**Reply by ludeeus** [reply]
I don't agree with this.
Instead we should not make them clickable while set state is "hidden"

**Reply by spacegaier** [reply]
> I don't agree with this. Instead we should not make them clickable while set state is "hidden"

Could you explain why you disagree? Otherwise a bit difficult to discuss 😜 

I find that 80 % of the time I am on that page I am there to change some states for testing. Personally I also never use the table but instead directly select the entity from the dropdown, so I am not a friend of the collapsed section to begin with, but understand that most regular users do not start with that.

BTW: In case we actually go that way and the entity IDs are no longer clickable, we should still visually highlight them for a nice, clear look (which currently the hyperlink does for "free").

**Reply by Mariusthvdb** [reply]
> I don't agree with this.
> 
> Instead we should not make them clickable while set state is "hidden"

That would be a huge breaking change, and not in a positive way either. Making entities non clickable is taking away from the main functionality of the states page in the first place imho. Making it conditional by first having to click Set state, adds even further to that.

Please don't.. 

If anything, just revert hiding it. People go to the states page to interact with entities. Anything causing a delay to do so after landing there is detrimental to that.

What would really be an improvement though is landing the dev tools on Template. That loads mostly immediately, while the States do take their time, especially on slower systems with a lot of states/entities .... see: https://github.com/home-assistant/frontend/discussions/9298

**Reply by ludeeus** [reply]
> Could you explain why you disagree

When you are not showing the "set state" section, there is no reason the entities should be clickable as you would only need that when the  "set state" section is open.

When you "now" click on an entity, the  "set state" section expands and pushes the UI.

As for "breaking change" no.. it's still there, it's still usable, you just have to go into  "set state" mode which makes sense.

**Reply by Mariusthvdb** [reply]
I can appreciate  the goal for the page to be as plain as can be. Having to go into "set state" mode each and every time we visit the page (personal use case) Id really love that state to be sticky if and when it would be changed as Ludeeus suggests. Would that be an option? Just as the template page remembers/stores the last template.
